cis194/week2/tien/homework2/src/LogAnalysis.hs

```diff
@@ -0,0 +1,47 @@
+{-# OPTIONS_GHC -Wall #-}
+
+module LogAnalysis where
+
+import           Log
+import           Text.Read (readMaybe)
+
+getHead :: [String] -> String
+getHead wordList =
+  case wordList of
+    x:_ -> x
+    _   -> ""
+
+getNthValue :: Int -> String -> Maybe Int
+getNthValue n str = readMaybe (getHead . drop (n-1) . words $ str) :: Maybe Int
+
+getNthTailConent :: Int -> String -> String
+getNthTailConent n str = unwords . drop n . words $ str
```

Comment:
The value returned by `getNthValue` and `getNthTailConent` are next to each other in a string.
This is a good indication that a pattern match on specific elements of that string should do the trick.
See comment on `parseMessage`.